store/sqlstore/role_supplier.go

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type Role struct {
+	Id            string
+	Name          string
+	DisplayName   string
+	Description   string
+	Permissions   string
+	SchemeManaged bool
+}
+
+func FromRole(role *model.Role) *Role {
+	permissionsMap := make(map[string]bool)
+	permissions := ""
+
+	for _, permission := range role.Permissions {
+		if _, ok := permissionsMap[permission]; !ok {
+			permissions += fmt.Sprintf(" %v", permission)
```

Comment:
I think we should serialize these a bit more robustly. Or validate them somewhere. If spaces aren't allowed, we need to add some checking. If they are, we need to add escaping.